tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1986,9 +1986,8 @@ GetCudnnConvolutionBackwardFilterAlgo(const CudnnHandle& cudnn,
 
 port::StatusOr<DeviceMemory<uint8>> AllocateCudnnConvolutionForwardWorkspace(
     Stream* stream, const CudnnHandle& cudnn,
-    const dnn::AlgorithmDesc& algorithm_desc,
-    const CudnnTensorDescriptor& input_nd, const CudnnFilterDescriptor& filter,
-    const CudnnConvolutionDescriptor& conv,
+    dnn::AlgorithmDesc& algorithm_desc, const CudnnTensorDescriptor& input_nd,
```

Comment:
algorithm_desc is an in-out parameter, so it should be a pointer and preferably put in the end (before scratch_allocator).

Same applies to other two functions.

Reply:
@timshen91 updated the PR according to the comment. please help take a look once again. thanks.